docs/notebooks/batch_optimization.pct.py

```diff
@@ -78,9 +75,9 @@ def build_model(data):
 from trieste.acquisition.rule import EfficientGlobalOptimization
 
 batch_ei_acq = BatchMonteCarloExpectedImprovement(sample_size=1000)
-points_chosen_by_batch_ei, _ = EfficientGlobalOptimization(
-    num_query_points=10, builder=batch_ei_acq.using(OBJECTIVE)
-).acquire(search_space, initial_data, models) # type: ignore
+batch_ei_acq_rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(
```

Comment:
I don't understand what this line does (`[Box]`)

Reply:
This is a (hopefully temporary) way to explain to the type checker what the search space looks like, since it can't figure it out by itself. This is why the line previously had a type: ignore comment. We do this elsewhere in the notebooks, though I would prefer to find a way to make this unnecessary.

Reply:
since these notebooks (and especially the intro one) audience include researchers, etc., maybe this would introduce too much confusion? while type: ignore is OK...

Reply:
Sure. I'll revert (and change the existing cases to type ignore too).